[next] <AsciiRenderer> (#1145)
threlte/threlte: apps/docs/src/content/reference/extras/ascii-renderer.mdx

name: <AsciiRenderer>
type: 'component'
componentSignature:
  {
    exports:
      [
        { name: 'start', type: '() => void', description: 'Manually start the render task' },
        { name: 'stop', type: '() => void', description: 'Manually stop the render task' },
        {
          name: 'getEffect',
          type: '() => THREE.AsciiEffect',
          description: 'Getter for the underlying AsciiEffect instance'
        }
      ],
    props:
      [
        {
          default: 'useThrelte().camera.current',
          description: 'the camera to use when renderering',
          name: 'camera',
          required: false,
          type: THREE.Camera
        },
        {
          default: 'useThrelte().scene',
          description: 'the scene to use when renderering',
          name: 'scene',
          required: false,
          type: THREE.Scene
        },
        {
          default: 'true',
          description: 'whether to automatically run the render task',
          name: 'autoRender',
          required: false,
          type: 'boolean'
        },
        {
          name: 'characters',
          default: "' .:-+*=%@#'",

>           description: 'character set used by the renderer. should be sorted from least to most opaque',

Reviewer: ```suggestion
          description: 'character set used by the renderer. Sorted left-to-right from least to most opaque',
```

<Example path="extras/ascii-renderer/basic" />

## Usage

Typically you'd use `<AsciiRenderer>` alongside your main `Scene` component. `<AsciiRenderer>` creates an absolutely postioned table element that is appended to the dom element of the `<Canvas>`. You may need to set the wrapping element's position to `relative`

```svelte
<div>
  <Canvas>
    <AsciiRenderer />
    <Scene />
  </Canvas>
</div>

<style>
  div {
    position: relative;
  }
</style>
```

## Characters

The `characters` prop should be sorted by ascending "opaqueness".

```svelte
<AsciiRenderer characters=" #" />
```

The example above uses a character set with two characters - ` ` and `#`.

## Colors

By default the renderer sets `options.color` to `false` and will only use the colors given by the `fgColor` and `bgColor` props. `fgColor` and `bgColor` can be any acceptable CSS color string. If your color contains an alpha component, make sure to set `options.alpha` to `true`.

If `options.color` is set to `true`, `fgColor` and `bgColor` will be ignored and the corresponding color of the scene will be used for each character.

```svelte
<AsciiRenderer options={{ color: true }} />
```

<Tip type="info">

>   Setting `options.color` to `true` really slows down performance but it can be used if you have a
>   static scene or only need to render the scene on demand.

Reviewer: ```suggestion
  Setting `options.color` to `true` slows down performance. Using a
  static scene or manually rendering can help.
  ```
